### ISSUE TITLE
review and update ComboBox.md

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/sun#465]
There have been many changes to ComboBox.js - #430, #314, #445,....  

ComboBox.md needs to reviewed and possibility updated, especially the a11y sections.

**Comment by terracoda** [comment]
@zepumph, please assign to me if you need me to help with any changes to ComboBox.md.

I'm not sure our a11y markdown docks have a place in the document yet that would describe that the widget (e.g. ComboBox) is made up of several different things. I think we will need a place in the document that lists out what makes up a ComboBox (e.g., perhaps list out the things that are included in this comment https://github.com/phetsims/sun/issues/445#issuecomment-455824952)

By the way, nice work everyone, in particular @pixelzoom and @zepumph for making a robust ComboBox.

**Comment by pixelzoom** [comment]
@terracoda said:
>  I think we will need a place in the document that lists out what makes up a ComboBox  ...

And it sounds like that may be different for JavaScript vs PDOM.   For example, ComboBox's JavaScript implementation is a container that is made up a button and a list box that contains items.  In the PDOM (currently), the button (not a "combo box") is the top-level thing, and the button has a list box.   To make things easier to understand, I think it's advantageous to keep the 2 representations as similar as possible.  But as @zepumph explained to me, sometimes there will be superior representations that different, and we want to have the flexibility to accommodate.   And differences should be documented in the associated .md file.

In the case of ComboBox... I wasn't involved in the a11y design for ComboBox, so I don't know if different representations are warranted in this case.  But we are using different representations, and it does take a little bit of work to understand the differences.  Those differences (what they are, why,...) should be documented in ComboBox.md.

**Comment by terracoda** [comment]
@pixelzoom, I actually do explain in the ComboBox.md file why we created and used the HTML structure that we use. You can find a brief explanation in the Accessibility section and just above the HTML code.

What kind of guidance or explanation do you think developers need to better understand how to implement the design pattern? 

In terms of user experience, the differences between a `combobox` and a `button+listbox` are quite subtle. The main differences will be for screen reader users due to how their AT reads out HTML structure. The actual keyboard interactions are almost identical if not identical to a `combobox` widget.

**Comment by terracoda** [comment]
@zepumph, have Home and End shortcut keys been implemented as an option for the accessible combobox? 

While the documentation says they are optional, the documentation also says that these shortcuts are highly recommended for lists with more than 5 items.

Molarity's combobox list has 9 items. I am not sure if there was a decision to not support these shortcuts at all or if this it would be worked on at a later date.

I'll mark this as a meeting item.

**Comment by terracoda** [comment]
@zepumph, the combobox implementation in Molarity is working really well. If you think we should handle the Home and End keys in a new issue, just say so.

**Comment by terracoda** [comment]
@zepumph, I discussed with @emily-phet, and we would like to have Home and End shortcuts in the long term. And we will see how we can loop this in for Molarity depending on time needed for this feature.

**Comment by zepumph** [comment]
>@zepumph, the combobox implementation in Molarity is working really well. If you think we should handle the Home and End keys in a new issue, just say so.

Sorry this issue was totally lost, this was implemented in https://github.com/phetsims/sun/issues/534. @terracoda is there anything else for this issue? I believe we can close.

**Comment by pixelzoom** [comment]
@terracoda could you please weigh in on whether there's anything else to do here? ComboBox has been done for quite awhile, and this is a "loose end" that needs to be tied up.

**Comment by terracoda** [comment]
Last I checked, the combobox was working fine in Molarity.
We just might revisit later if we want to make it behave differently on mobile. I think there is a different issue for that.

**Comment by terracoda** [comment]
And I will add, the combobox works nicely on mobile, but the popped up list stays open if nothing is selected and the user moves their cursor. This is not a barrier...just something to work out in a different issue. 

I think this one can be closed if the review is all done.
